src/main/java/com/google/sps/data/ComputeAPIClient.java

```diff
@@ -0,0 +1,225 @@
+          credential = credential.createScoped(scopes);
+        }
+        HttpRequestInitializer requestInitializer = 
+          new HttpCredentialsAdapter(credential);
+        // Create Compute Engine object.
+        compute = new Compute.Builder
+          (httpTransport, JSON_FACTORY, requestInitializer)
+          .setApplicationName(APPLICATION_NAME)
+          .build();
+        Operation op = null;
+        if (operation == "createInstance") {
+          op = createInstance(instanceName, password.get());
+        }
+        else if (operation == "deleteInstance") {
+          op = deleteInstance(instanceName);
+        }
+        else if (operation == "stopInstance") {
+          op = stopInstance(instanceName);
+        }
+        else if (operation == "restartInstance") {
+          op = restartInstance(instanceName, password.get());
+        }
+        else {
+          System.out.println("Operation not found!");
+        }
+        // Call Compute Engine API operation and poll for operation completion 
+        // status
+        System.out.println("Waiting for operation completion...");
+        Operation.Error error = blockUntilComplete
+          (compute, op, OPERATION_TIMEOUT_MILLIS);
+        if (error == null) {
+          System.out.println("Success!");
+        } else {
+          System.out.println(error.toPrettyString());
+        }
+      } catch (IOException e) {
+          System.out.println("Got error!");
+          System.err.println(e.getMessage());
+      } catch (Throwable t) {
+          t.printStackTrace();
+        }
+    }
+    
+  // [START create_instances]
+  public Operation createInstance(String instanceName, String password)
+    throws IOException {
+      Instance instance = new Instance();
+      instance.setName(instanceName);
+      instance.setMachineType(String.format(
+        "projects/taniecer-step-2020/zones/us-central1-f/machineTypes/n1-standard-1",
+        PROJECT_ID, ZONE_NAME));
+      CustomerEncryptionKey key = new CustomerEncryptionKey();
+      key.setRawKey(password);
+      // Add Network Interface to be used by VM Instance.
+      NetworkInterface ifc = new NetworkInterface();
+      ifc.setNetwork(String.format(
+        "https://www.googleapis.com/compute/v1/projects/%s/global/networks/default",
+        PROJECT_ID));
+      List<AccessConfig> configs = new ArrayList<>();
+      AccessConfig config = new AccessConfig();
+      config.setType(NETWORK_INTERFACE_CONFIG);
+      config.setName(NETWORK_ACCESS_CONFIG);
+      configs.add(config);
+      ifc.setAccessConfigs(configs);
+      instance.setNetworkInterfaces(Collections.singletonList(ifc));
+      AttachedDisk disk = new AttachedDisk();
+      disk.setBoot(true);
+      disk.setAutoDelete(true);
+      disk.setType("PERSISTENT");
+      AttachedDiskInitializeParams params = new AttachedDiskInitializeParams();
+      params.setDiskName(instanceName);
+      params.setSourceImage(SOURCE_IMAGE_PREFIX + SOURCE_IMAGE_PATH);
```

Comment:
were you able to set the source image as the custom image? the SOURCE_IMAGE_PATH would be cdesir-step-2020/global/images/vmn-image

Reply:
I don't have access to the custom image so for now it is a random image.